examples/ecies-example.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <mococrw/ecies.h>
+#include <mococrw/hash.h>
+#include <mococrw/util.h>
+#include <mococrw/openssl_wrap.h>
+
+#include <iostream>
+
+using namespace mococrw;
+
+struct EciesEncryptResult {
+    std::vector<uint8_t> ephKey;
+    std::vector<uint8_t> ciphertext;
+    std::vector<uint8_t> mac;
+};
+
+static const std::string pubKeyPem =R"(
+-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAELGj/fqXab+amxDOcbDWMArPBsUMOPons
+9NePVyS9CbIkI8e3nPYi3ytHjJm03M22vM5R4XAxI9cMv5biylFJW0HBlyf/cZTO
+xOVCTvRDUHLGTdGXNlz74YtWLF+CMX5A
+-----END PUBLIC KEY-----
+)";
+
+static const std::string privKeyPem =R"(
+-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDDP/GdwmJa6KFj/R7QJRi7wNNG+viSMos2B+4zIi7fO5BUpIMZObh6/
+ujDnPKFx4SugBwYFK4EEACKhZANiAAQsaP9+pdpv5qbEM5xsNYwCs8GxQw4+iez0
+149XJL0JsiQjx7ec9iLfK0eMmbTczba8zlHhcDEj1wy/luLKUUlbQcGXJ/9xlM7E
+5UJO9ENQcsZN0Zc2XPvhi1YsX4IxfkA=
+-----END EC PRIVATE KEY-----
+)";
+
+static const auto digest = DigestTypes::SHA512;
```

Comment:
This would require code duplication as these constants are used for encryption and decryption.
Shall I leave it there or duplicate the code for en- and decrypting?